### PR TITLE
conformance-k8s-kind: Use Helm mode cilium-cli

### PR DESCRIPTION
**Pull request by michi-covalent**
Ref: https://github.com/cilium/cilium-cli#experimental-helm-installation-mode
Ref: #25156

### CODE REVIEW

File: .github/workflows/conformance-k8s-kind.yaml [hunk 1]
```diff
@@ -147,14 +148,8 @@ jobs:
             --helm-set=cni.chainingMode=portmap \
             --helm-set-string=kubeProxyReplacement=strict \
             --helm-set=sessionAffinity=true \
-            --rollback=false \
-            --config monitor-aggregation=none \
-            --version="
-          HUBBLE_ENABLE_DEFAULTS="--chart-directory=install/kubernetes/cilium \
-            --relay-image=quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/hubble-relay-ci:${SHA} \
-            --relay-version=${SHA}"
+            --helm-set=bpf.monitorAggregation=none"
```
Comment (sayboras) [inline .github/workflows/conformance-k8s-kind.yaml:151]: you also need to set hubble relay enabled flag as well.

File: .github/workflows/conformance-k8s-kind.yaml [hunk 2]
```diff
@@ -26,6 +26,7 @@ env:
   # renovate: datasource=github-releases depName=cilium/cilium-cli
   cilium_cli_version: v0.14.5
   cilium_cli_ci_version:
+  CILIUM_CLI_MODE: helm
```
Comment (sayboras) [inline .github/workflows/conformance-k8s-kind.yaml:29]: I actually try with global env like this, but i faced a couple of issue like flag abc is not supported for cilium cli command (other than cilium install), not sure if it works here, and for all cases.

Comment (michi-covalent) [reply .github/workflows/conformance-k8s-kind.yaml:29]: yeah let's see if i'm lucky enough.

File: .github/workflows/conformance-k8s-kind.yaml (hunk 1 above)
Comment (michi-covalent) [reply .github/workflows/conformance-k8s-kind.yaml:151]: this workflow doesn't use hubble relay right? 🤔 https://github.com/cilium/cilium/blob/main/.github/workflows/conformance-k8s-kind.yaml

Comment (sayboras) [reply .github/workflows/conformance-k8s-kind.yaml:151]: you are right, seems like the hubble_enable_defaults env is not used :face_exhaling:

**Review by sayboras — APPROVED** [review]
:ship:

**Review by aanm — APPROVED** [review]

**Pull request merged**